reference/forms/types/choice.rst

```diff
@@ -83,40 +87,98 @@ code, while the **key** is what will be shown to the user.
     and will be removed in 3.0. To read about the old API, read an older version of
     the docs.
 
+.. note::
+
+    Pre selected choices will depend on the **data** passed to the field and
+    the values of the ``choices`` option. However submitted choices will depend
+    on the **string** matching the **choice**. In the example above, the default
+    values are incrementing integers because ``null`` cannot be casted to string.
+    You should consider it as well when dealing with ``empty_data`` option::
+
+        $builder->add('isAttending', 'choice', array(
+            'choices'  => array(
+                'Maybe' => null,
+                'Yes' => true,
+                'No' => false,
+            ),
+            'choices_as_values' => true,
+            'data' => true, // pre selected choice
```

Comment:
This should be a bad practice. The reader could interpret that is a way to set default value and it isn't. As you know, if an object is bound to the form, the `data` option overrides this default value.

I suggest you to use `$this->createFormBuilder(array('isAttending' => true))` from a controller for this sample. WDYT?